src/NUnitFramework/framework/Constraints/MsgUtils.cs

```diff
@@ -94,6 +96,55 @@ static MsgUtils()
             AddFormatter(next => val => val is Array valArray ? FormatArray(valArray) : next(val));
         }
 
+        /// <summary>
+        /// Try to format the properties of an object as a string.
+        /// </summary>
+        /// <param name="val">The object to format.</param>
+        /// <returns>Formatted string for all properties.</returns>
+        public static string FormatValueProperties(object? val)
```

Comment:
nitpick: Was there a use case for making this public, or could it be made private to maximize our ability to evolve it without breaking others?

For example, there could be a time in the future where we'd want to pass in a precomputed collection of properties from elsewhere in the framework, or perhaps allowing this to recurse to child objects with a configurable "depth".

Reply:
> nitpick: Was there a use case for making this public, or could it be made private to maximize our ability to evolve it without breaking others?

It cannot be private as it is called from another class (`EqualConstraintResult`). I didn't want to introduce it in the formatter call chain for _all_ values as that will have side effects where properties are not to be printed, like DbContext.

The whole class `MsgUtils` is `internal` and therefore we can change the signature without breaking external code.

Reply:
Oh! I had meant internal, but has missed that MsgUtils itself was already internal. Sorry about that